test/server/controller.spec.js

```diff
@@ -1100,101 +1441,126 @@ describe('redirection when running behind a proxy', () => {
-      headers:    {'X-Forwarded-Host': 'myserver.com:3456'},
-      method:    'GET'};
-    let req = http.request(options);
-    req.on('response', (res) => {
-      var body = '';
-      expect(res.statusCode).toEqual(200);
-      res.on('data', (d) => { body += d;});
-      res.on('end', () => {
-        expect(JSON.parse(body)).toEqual({
-          htsget: {
-            format: 'BAM',
-            urls: [{'url': url}]
-          }
+    it(`Redirection to one of known proxies for ${optionsList.method}`, (done) => {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      obj.path = serverPath;
+      obj.headers =  {'X-Forwarded-Host': 'myserver.com:3456'};
+      let url = `http://myserver.com:3456/path3/sample?accession=${id}&format=BAM`;
+      let req = http.request( obj );
+      req.on('response', (res) => {
+        var body = '';
+        expect(res.statusCode).toEqual(200);
+        res.on('data', (d) => { body += d;});
+        res.on('end', () => {
+          expect(JSON.parse(body)).toEqual({
+            htsget: {
+              format: 'BAM',
+              urls: [{'url': url}]
+            }
+          });
+          done();
         });
-        done();
       });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam", "accession":id}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
-    req.end();
-  });
 
-  it('Redirection to one of known proxies', (done) => {
-    let url = `http://myserver.com/path1/path2/sample?accession=${id}&format=BAM`;
-    let options = {
-      socketPath: socket,
-      path:       serverPath,
-      headers:    {'X-Forwarded-Host': 'myserver.com'},
-      method:    'GET'};
-    let req = http.request(options);
-    req.on('response', (res) => {
-      var body = '';
-      expect(res.statusCode).toEqual(200);
-      res.on('data', (d) => { body += d;});
-      res.on('end', () => {
-        expect(JSON.parse(body)).toEqual({
-          htsget: {
-            format: 'BAM', urls: [{'url': url}]
-          }
+    it(`Redirection to one of known proxies for ${optionsList.method}`, (done) => {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      obj.path = serverPath;
+      obj.headers =  {'X-Forwarded-Host': 'myserver.com'};
+      let url = `http://myserver.com/path1/path2/sample?accession=${id}&format=BAM`;
+      let req = http.request( obj );
+      req.on('response', (res) => {
+        var body = '';
+        expect(res.statusCode).toEqual(200);
+        res.on('data', (d) => { body += d;});
+        res.on('end', () => {
+          expect(JSON.parse(body)).toEqual({
+            htsget: {
+              format: 'BAM', urls: [{'url': url}]
+            }
+          });
+          done();
         });
-        done();
       });
-    });
-    req.end();
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam", "accession":id}), () => {
```

Comment:
I was not expecting you to need to specify a `accession` here. Can you delete it and see if test still runs?